### PR TITLE
sanity powered blog with mdx support

### PR DESCRIPTION
**Pull request by joelhooks**
Following Sanity examples (kitchen sink, nextjs blog) this adds Sanity backed blog content at `/blog` to replace a lot of the content at `/learn` that would qualify as a traditional "blog post"

### CODE REVIEW

File: studio/deskStructure.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    'person',
+    'software-library',
+    'post',
+    'page',
+    'siteSettings',
+    'author',
+    'category',
+  ].includes(listItem.getId())
+
+export default () =>
```
Comment (joelhooks) [inline studio/deskStructure.js:18]: this is where you can start to consider custom publishing workflows

File: studio/schemas/documents/post.js [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+          type: 'reference',
+          to: {
+            type: 'category',
+          },
+        },
+      ],
+    },
+    {
+      name: 'body',
+      type: 'markdown',
```
Comment (joelhooks) [inline studio/schemas/documents/post.js:66]: I tried to enjoy their "portable text" but MDX has my heart here for sure.

File: src/pages/blog/[slug].tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
+import * as React from 'react'
+import groq from 'groq'
+import {sanityClient} from 'utils/sanity-client'
+import imageUrlBuilder from '@sanity/image-url'
+import mdxComponents from 'components/mdx'
+import renderToString from 'next-mdx-remote/render-to-string'
```
Comment (joelhooks) [inline src/pages/blog/[slug].tsx:6]: `next-mdx-remote` has limitations, but works very well for this use case. Primarily I believe you can't import so all components need to be pre-defined.

Comment (kentcdodds) [reply src/pages/blog/[slug].tsx:6]: If that bugs you, checkout mdx-bundler by yours truly 😉

**Comment by joelhooks** [comment]
This looks awesome. Any Next examples yet? Would you be up for showing me how to convert (maybe as a stream?)

**Comment by kentcdodds** [comment]
I've not tried it with next yet, though I'm sure it works. A stream would be fun!

**Pull request merged**